File: c/1394.c
int findLucky(int* arr, int n) {
    int f[501] = {0};
    for (int i = 0; i < n; i++)
        ++f[arr[i]];
    for (int i = 500; i >= 1; i--)
        if (i == f[i])
            return i;
    return -1;
}

File: python3/1394.py
class Solution:
    def findLucky(self, arr: List[int]) -> int:
        f = [0] * 501
        for i in arr:
            f[i] += 1
        for i in range(500, 0, -1):
            if i == f[i]:
                return i
        return -1
